### PR TITLE
Fix sublayer

### PR DESCRIPTION
**Pull request by JiabinYang**
### PR types

Bug fixes
### PR changes

APIs
### Describe

This PR fixes issue of https://github.com/PaddlePaddle/Paddle/issues/31610

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_layers.py [hunk 1]
```diff
@@ -3718,6 +3718,35 @@ def test_set_train_eval_in_static_mode(self):
...
+        return temp
+
+
+class MySuperLayer(paddle.nn.Layer):
+    def __init__(self):
+        super(MySuperLayer, self).__init__()
+        self._mylayer = MyLayer()
+
+    def forward(self, input):
+        temp = self.mylayer(input)
```
Comment (XiaoguangHu01) [inline python/paddle/fluid/tests/unittests/test_layers.py]: mylayer -> _mylayer ?

File: python/paddle/fluid/tests/unittests/test_layers.py [hunk 2]
```diff
@@ -3718,6 +3718,35 @@ def test_set_train_eval_in_static_mode(self):
...
+        return temp
+
+
+class TestSubLayerCount(unittest.TestCase):
+    def test_sublayer(self):
+        with fluid.dygraph.guard():
+            mySuperlayer = MySuperLayer()
+            self.assertTrue(len(mySuperlayer.sublayers()) == 3)
```
Comment (XiaoguangHu01) [inline python/paddle/fluid/tests/unittests/test_layers.py:3747]: 测试下include_self=False的情况

Comment (JiabinYang) [reply python/paddle/fluid/tests/unittests/test_layers.py:3747]: 这个参数默认就是falsed

File: python/paddle/fluid/tests/unittests/test_layers.py (hunk 1 above)
Comment (JiabinYang) [reply python/paddle/fluid/tests/unittests/test_layers.py]: done, thx

File: python/paddle/fluid/tests/unittests/test_layers.py (hunk 2 above)
Comment (JiabinYang) [reply python/paddle/fluid/tests/unittests/test_layers.py:3747]: 增加了测试`include_self=True`的测试

**Review by saxon-zh — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**